inputremapper/injection/mapping_handlers/rel_to_rel_handler.py

```diff
@@ -0,0 +1,188 @@
+    _moving: asyncio.Event  # event to notify the _recenter_loop
+
+    _remainder: float
+
+    def __init__(
+        self,
+        combination: EventCombination,
+        mapping: Mapping,
+        **_,
+    ) -> None:
+        super().__init__(combination, mapping)
+
+        self._remainder = 0
+
+        # find the input event we are supposed to map. If the input combination is
+        # BTN_A + REL_X + BTN_B, then use the value of REL_X for the transformation
+        input_event = mapping.find_analog_input_event(type_=EV_REL)
+        assert input_event is not None
+        self._input_event = input_event
+
+        # - If rel_x is mapped to rel_y, it will transform it to between 0 and 1,
+        # and then scale it back to exactly its original value.
+        # - If rel_x is mapped to rel_wheel, and the mouse is moved in a normal
+        # tempo, then the wheel should move in a normal tempo as well.
+        # -> So the same speed is used as max_ and for scaling.
+        if self._input_event.is_wheel_event:
+            max_ = self.mapping.rel_wheel_speed
+        elif self._input_event.is_hi_res_wheel_event:
+            max_ = self.mapping.rel_hi_res_wheel_speed
+        else:
+            max_ = self.mapping.rel_xy_speed
+
+        self._transform = Transformation(
+            max_=max_,
+            min_=-max_,
+            deadzone=self.mapping.deadzone,
+            gain=self.mapping.gain,
+            expo=self.mapping.expo,
+        )
+
+    def __str__(self):
+        return f"RelToRelHandler for {self._input_event} <{id(self)}>:"
+
+    def __repr__(self):
+        return self.__str__()
+
+    @property
+    def child(self):  # used for logging
+        return f"maps to: {self.mapping.output_code} at {self.mapping.target_uinput}"
+
+    def _should_map(self, event):
+        """Check if this input event is relevant for this handler."""
+        if self._input_event.is_wheel_event and is_wheel(event):
+            return True
+
+        if event.type_and_code == (self._input_event.type, self._input_event.code):
+            return True
+
+        return False
+
+    def notify(
+        self,
+        event: InputEvent,
+        source: evdev.InputDevice,
+        forward: evdev.UInput = None,
+        supress: bool = False,
+    ) -> bool:
+        if not self._should_map(event):
+            return False
+
+        try:
+            self._write(self._transform(event.value))
+            return True
+        except (exceptions.UinputNotAvailable, exceptions.EventNotHandled):
+            return False
+
+    def reset(self) -> None:
+        pass
+
+    def _write(self, value: float) -> None:
+        """Inject."""
+        # value is between 0 and 1, scale up
+        if self.mapping.is_wheel_output():
+            scaled = value * self.mapping.rel_wheel_speed
+        elif self.mapping.is_high_res_wheel_output():
+            scaled = value * self.mapping.rel_hi_res_wheel_speed
```

Comment:
There was this https://github.com/sezanzeb/input-remapper/issues/354#issuecomment-1079728380 issue where wayland can not handle HI_RES wheel events. So in the AbsToRelHandler I made sure to always send both events in the correct ratio 1/120.

If the mapping output is `WHEEL_HI_RES` the I would send the `WHEEL_HI_RES` event an accumulate output-value/120 until it is >= 1 and then also send `WHEEL` event.
If the mapping output is `WHEEL` I would always send the `WHEEL` event and 120*value as `WHEEL_HI_RES`.

Reply:
thanks, I implemented it